src/test/kotlin/com/stefan_grafberger/streamdq/checks/row/RowValueIsNonNegativeTest.kt

```diff
@@ -0,0 +1,79 @@
+/**
+ * Licensed to the University of Amsterdam (UvA) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The UvA licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.stefan_grafberger.streamdq.checks.row
+
+import com.stefan_grafberger.streamdq.TestUtils
+import com.stefan_grafberger.streamdq.VerificationSuite
+import com.stefan_grafberger.streamdq.data.TestDataUtils
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import java.math.BigDecimal
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class RowValueIsNonNegativeTest {
+    @Test
+    fun `test RowValueIsNonNegative works`() {
+        val (env, rawStream) = TestDataUtils.createEnvAndGetClickStream()
+
+        val rowLevelCheck = RowLevelCheck()
+                .isNonNegative("timestamp")
+        val verificationResult = VerificationSuite()
+                .onDataStream(rawStream, env.config)
+                .addRowLevelCheck(rowLevelCheck)
+                .build()
+
+        val result = TestUtils.collectRowLevelResultStreamAndAssertLen(verificationResult, rowLevelCheck, 10)
+
+        TestUtils.assertRowLevelConstraintResults(
+                result, rowLevelCheck, 0,
+                arrayOf(true, true, true, true, true, true, true, true, true, true)
+        )
+
+    }
+
+    @Test
+    fun testMapWhenFieldValueIsDigDecimalExpectCastCorrect() {
+        //arrange
+        val (env, rawStream) = TestDataUtils.createEnvAndGetClickStream()
```

Comment:
Would it be possible to use test data here that contains a big decimal attribute but with values that are higher than Double.MAX_POSITIVE_VALUE (or whatever it is called in Kotlin)?

Reply:
Tried, it will fail with an exception called `NumberFormatException`, looks the biggest Double value which allowed to transfer to BigDecimal is Double.MAX_POSITIVE_VALUE

Reply:
pushed this commit https://github.com/stefan-grafberger/StreamDQ/pull/5/commits/e58ea0349d6b5318f46320eee96c5682c090adab

Reply:
Sorry if I wasn't clear, but this exact situation with values that are bigger than Double.MAX_POSITIVE_VALUE is the problem I meant in the original implementation in the RowValueIsNonNegative map function. You can create BigDecimals that are larger than Double.MAX_POSITIVE_VALUE, e.g., with the constructor [BigDecimal](https://docs.oracle.com/javase/8/docs/api/java/math/BigDecimal.html#BigDecimal-java.math.BigInteger-int-)([BigInteger](https://docs.oracle.com/javase/8/docs/api/java/math/BigInteger.html) unscaledVal, int scale). And these sort of numbers are exactly when the previous map function implementation would have thrown exceptions. The roundtrip from BigDecimal to Double to BigDecimal would not have worked. But I think with the new implementation everything should work hopefully :) 

Still, could you maybe update this test to use a number that is bigger than Double.MAX_POSITIVE_VALUE, e.g., by using this constructor: [BigDecimal](https://docs.oracle.com/javase/8/docs/api/java/math/BigDecimal.html#BigDecimal-java.math.BigInteger-int-)([BigInteger](https://docs.oracle.com/javase/8/docs/api/java/math/BigInteger.html) unscaledVal, int scale)?

Reply:
Yes Stefan, done and this time it works perfectly with the new approach